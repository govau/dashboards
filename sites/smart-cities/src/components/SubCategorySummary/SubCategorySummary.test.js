import React from 'react';
import { shallow } from 'enzyme';
import CategoryBanner from './SubCategorySummary';

jest.mock('../IndicatorCard/IndicatorCard', () => 'IndicatorCard');
jest.mock('../Icon/Icon', () => 'Icon');

const defaultProps = {
  categoryColorName: 'primary',
  highlightColorLight: 'JOBS_020',
  highlightColorDark: 'JOBS_600',
  name: 'Mock sub category',
  iconId: 'infrastructureGettingToWork',
  summaryIndicatorIds: [
    'population',
  ],
  cities: [
    {
      id: 'perth',
      name: 'Perth',
      indicators: {}
    },
    {
      id: 'sydney',
      name: 'Sydney',
      indicators: {}
    },
  ],
  charts: [1, 2, 3], // the charts array is only used to get the number of charts
};

it('should match Snapshot', () => {
  const component = shallow(
    <CategoryBanner {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should show the correct number of charts in the link to charts', () => {
  const component = shallow(
    <CategoryBanner {...defaultProps} />
  );

  expect(component.find('.linkText').text()).toBe('View all 3 charts');
});

it('should show the non-plural text to charts when only one chart', () => {
  const component = shallow(
    <CategoryBanner
      {...defaultProps}
      charts={[1]}
    />
  );

  expect(component.find('.linkText').text()).toBe('View 1 chart');
});
