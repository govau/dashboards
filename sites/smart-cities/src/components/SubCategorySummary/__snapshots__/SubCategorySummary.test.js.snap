// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div style={{...}} className=\\"wrapper\\">
  <div className=\\"container\\">
    <div className=\\"textWrapper\\">
      <Icon className=\\"iconWrapper\\" icon=\\"infrastructureGettingToWork\\" color=\\"JOBS_600\\" size={80} />
      <h3 className=\\"title\\">
        Mock sub category
      </h3>
    </div>
    <div className=\\"cardRow\\">
      <div className=\\"cardWrapper\\">
        <IndicatorCard className=\\"card\\" indicatorId=\\"population\\" color=\\"JOBS_600\\" colorName=\\"primary\\" cities={{...}} city={[undefined]} />
        <Pill className=\\"linkWrapper\\" height={32}>
          <a href=\\"#sc-mock-sub-category\\">
            <span className=\\"linkText\\">
              View all 3 charts
            </span>
            <Icon className=\\"linkIcon\\" icon=\\"rightArrowInCircle\\" size={20} color=\\"PRIMARY_900\\" />
          </a>
        </Pill>
      </div>
    </div>
  </div>
</div>"
`;
